File: src/constants/categories.ts
export interface Category {
  name: string;
  bucket: "Needs" | "Wants" | "Savings";
  icon: string;
  color: string;
}

export const CATEGORIES: Category[] = [
  // Needs (70%)
  { name: "Food", bucket: "Needs", icon: "🍔", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },
  { name: "Rent", bucket: "Needs", icon: "🏠", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },
  { name: "Transport", bucket: "Needs", icon: "🚗", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },
  { name: "Bills", bucket: "Needs", icon: "💡", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },
  { name: "Education", bucket: "Needs", icon: "📚", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },
  { name: "Medical", bucket: "Needs", icon: "❤️", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },
  { name: "Utilities", bucket: "Needs", icon: "🔌", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },
  { name: "Mobile & Internet", bucket: "Needs", icon: "📱", color: "bg-emerald-500/10 text-emerald-600 border-emerald-500/20" },

  // Wants (20%)
  { name: "Entertainment", bucket: "Wants", icon: "🎬", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },
  { name: "Shopping", bucket: "Wants", icon: "🛍️", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },
  { name: "Coffee & Cafes", bucket: "Wants", icon: "☕", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },
  { name: "Dining Out", bucket: "Wants", icon: "🍽️", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },
  { name: "Gaming", bucket: "Wants", icon: "🎮", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },
  { name: "Gifts", bucket: "Wants", icon: "🎁", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },
  { name: "Travel", bucket: "Wants", icon: "✈️", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },
  { name: "Fun", bucket: "Wants", icon: "🎉", color: "bg-[#FF5CA8]/10 text-[#FF5CA8] border-[#FF5CA8]/20" },

  // Savings (10%)
  { name: "Savings/Investment", bucket: "Savings", icon: "💰", color: "bg-[#8B5CF6]/10 text-[#8B5CF6] border-[#8B5CF6]/20" }
];

export function getBucketForCategory(category: string): "Needs" | "Wants" | "Savings" {
  const found = CATEGORIES.find(c => c.name.toLowerCase() === category.toLowerCase());
  return found ? found.bucket : "Needs";
}

File: android-backend/app/categories.py
"""
The 17 categories the frontend actually sends, and their bucket mapping.

This mirrors src/constants/categories.ts in the frontend exactly — same names,
same 70/20/10 buckets. Keeping category->bucket honest is the whole point of
the 70/20/10 analysis: if "Coffee & Cafes" (a Want) could silently become a
Need, the comparison becomes noise.
"""
from enum import Enum


class Bucket(str, Enum):
    NEEDS = "Needs"
    WANTS = "Wants"
    SAVINGS = "Savings"


class Category(str, Enum):
    FOOD = "Food"
    RENT = "Rent"
    TRANSPORT = "Transport"
    BILLS = "Bills"
    EDUCATION = "Education"
    MEDICAL = "Medical"
    UTILITIES = "Utilities"
    MOBILE_INTERNET = "Mobile & Internet"
    ENTERTAINMENT = "Entertainment"
    SHOPPING = "Shopping"
    COFFEE_CAFES = "Coffee & Cafes"
    DINING_OUT = "Dining Out"
    GAMING = "Gaming"
    GIFTS = "Gifts"
    TRAVEL = "Travel"
    FUN = "Fun"
    SAVINGS_INVESTMENT = "Savings/Investment"


CATEGORY_TO_BUCKET: dict[Category, Bucket] = {
    Category.FOOD: Bucket.NEEDS,
    Category.RENT: Bucket.NEEDS,
    Category.TRANSPORT: Bucket.NEEDS,
    Category.BILLS: Bucket.NEEDS,
    Category.EDUCATION: Bucket.NEEDS,
    Category.MEDICAL: Bucket.NEEDS,
    Category.UTILITIES: Bucket.NEEDS,
    Category.MOBILE_INTERNET: Bucket.NEEDS,
    Category.ENTERTAINMENT: Bucket.WANTS,
    Category.SHOPPING: Bucket.WANTS,
    Category.COFFEE_CAFES: Bucket.WANTS,
    Category.DINING_OUT: Bucket.WANTS,
    Category.GAMING: Bucket.WANTS,
    Category.GIFTS: Bucket.WANTS,
    Category.TRAVEL: Bucket.WANTS,
    Category.FUN: Bucket.WANTS,
    Category.SAVINGS_INVESTMENT: Bucket.SAVINGS,
}

# 70/20/10 targets — used by the summary endpoint's budget math.
BUCKET_TARGETS: dict[Bucket, float] = {
    Bucket.NEEDS: 0.70,
    Bucket.WANTS: 0.20,
    Bucket.SAVINGS: 0.10,
}

# The order the frontend renders categories in the analysis screen.
CATEGORY_ORDER: list[Category] = [
    Category.FOOD,
    Category.RENT,
    Category.TRANSPORT,
    Category.BILLS,
    Category.EDUCATION,
    Category.MEDICAL,
    Category.UTILITIES,
    Category.MOBILE_INTERNET,
    Category.ENTERTAINMENT,
    Category.SHOPPING,
    Category.COFFEE_CAFES,
    Category.DINING_OUT,
    Category.GAMING,
    Category.GIFTS,
    Category.TRAVEL,
    Category.FUN,
    Category.SAVINGS_INVESTMENT,
]


def get_bucket_for_category(category: Category) -> Bucket:
    """Look up the bucket for a category. Raises if it isn't one of the 17."""
    return CATEGORY_TO_BUCKET[category]
